Tests/FluentAssertions.Specs/Specialized/DelegateAssertionSpecs.cs

```diff
@@ -7,37 +7,52 @@ namespace FluentAssertions.Specs.Specialized;
 
 public class DelegateAssertionSpecs
 {
-    [Fact]
-    public void When_injecting_a_null_extractor_it_should_throw()
+    public class Throw
     {
-        // Arrange
-        Action subject = () => { };
-
-        // Act
-        Func<ActionAssertions> act = () => new ActionAssertions(subject, extractor: null);
+        [Fact]
+        public void Allow_additional_assertions_on_return_value()
+        {
+            // Arrange
+            var exception = new Exception("foo");
+            Action subject = () => throw exception;
 
-        // Act
-        act.Should().ThrowExactly<ArgumentNullException>()
-            .WithParameterName("extractor");
+            // Act / Assert
+            subject.Should().Throw<Exception>()
+                .Which.Message.Should().Be("foo");
+        }
     }
 
-    [Fact]
-    public void When_injecting_a_null_clock_it_should_throw()
+    public class ThrowExactly
     {
-        // Arrange
-        Action subject = () => { };
-        IExtractExceptions extractor = new AggregateExceptionExtractor();
+        [Fact]
+        public void When_injecting_a_null_extractor_it_should_throw()
+        {
+            // Arrange
+            Action subject = () => { };
 
-        // Act
-        Func<ActionAssertions> act = () => new ActionAssertions(subject, extractor, clock: null);
+            // Act
+            Func<ActionAssertions> act = () => new ActionAssertions(subject, extractor: null);
```

Comment:
🔧 `ActionAssertions` is not supposed to be used directly and thus should not be directly tests.

Reply:
I am not sure, I understand your issue. The two tests that use the `ActionAssertions` class existed previously. I only moved them to the `ThrowExactly` nested class.
Do you want me to remove them? Don't they test a valid scenario?

Reply:
Those where never supposed to be tested directly. So, if we move forward with this PR, I believe we should fix that.

Reply:
I removed the two tests.

Reply:
I think we need to take a different route. Instead of deleting those tests (and reducing the coverage), we should see if we can achieve the following:
1. Hide the constructors by making them `internal`
2. Rewrite the tests so they always use `Should` 
3. Maybe mark the `AssertionExtensions` as `sealed`

What do you think @jnyrup ?

Reply:
Please note, that the constructors of all "*Assertions" classes I checked were public, so I think it would be inconsistent to change these to internal.

Regarding rewriting the tests:
- The test checking for a `null` clock is simple to rewrite
- The second test checking for `null` on the `extractor` is more difficult. All internally used constructor calls use the static [`Extractor`](https://github.com/fluentassertions/fluentassertions/blob/develop/Src/FluentAssertions/AssertionExtensions.cs#L33) field in `AssertionExtensions` which can not be `null`.

Reply:
My rationale was that `AssertionExtensions` is not supposed to be used outside the library itself. Oftentimes we keep the constructors `public` or `protected` to allow consumer to build extensions. But I can't think of an example where that would be necessary.